### ISSUE TITLE
[Regression]: 0.11.0 , nested entity update support broke readMany

### ISSUE THREAD

**Issue by chihabhajji** [woowabros/nestjs-library-crud#463]
Hello, as always, thank you for the amazing lib,

Last week's feature #389 to be able to support table inheritence has rendered readMany and readOne to break, the lib wasnt able to read the metadata of the entity thus making it break them, reverted to 0.10.2 and everything works...

I can provider a minimal reproduction repo if necessary

**Comment by chihabhajji** [comment]
yes, ill start working on a minimal reproduction after i finish work

**Comment by chihabhajji** [comment]
https://github.com/chihabhajji/woowabros-reproduction

![image](https://github.com/woowabros/nestjs-library-crud/assets/35853327/b0b45f13-a37c-4add-9bd9-aaa296ade601)

swagger available at /docs

**Comment by chihabhajji** [comment]
side note, the same works if i down grade to 0.10 but then i lose the ability to fetch nested entities

**Comment by jiho-kr** [comment]
Okay, I`ll look into the information you gave.

**Comment by jiho-kr** [comment]
I'm trying to look at this, but I got covid-19. 
i waiting for condition to recover.

**Comment by chihabhajji** [comment]
no worries, get well soon, your health is the priority

**Comment by chihabhajji** [comment]
closing, fixed by using class-validator on my entities instead of @nestjs/class-validator, kindly add a PSA in the readme for new users

**Comment by chihabhajji** [comment]
my bad, i down graded to 0.10.2 when i did that so it worked

**Comment by jiho-kr** [comment]
i fixed it (https://github.com/woowabros/nestjs-library-crud/pull/470)

**Comment by chihabhajji** [comment]
you're the best, also i hope your health is better because i also caught the rona 🤦

**Comment by chihabhajji** [comment]
i tried the branch, the regression seems to be fixed, but extending a base entity seems to break (the base entity has created_at deleted_at and updated_at timestamps)
![image](https://github.com/woowabros/nestjs-library-crud/assets/35853327/0dd839c5-3e95-450e-948a-c018ae95f09a)

**Comment by jiho-kr** [comment]
It seems like failure for `softDelete` condition.

Would you like to check if you have defined `@DeleteDateColumn` on entity?

**Comment by chihabhajji** [comment]
`ts
export abstract class BaseFinvoEntity extends BaseEntity {
    @IsDateString(undefined, {
        groups: [GROUP.READ_MANY, GROUP.READ_ONE],
    })
    @IsOptional({ groups: [GROUP.READ_MANY, GROUP.READ_ONE] })
    @Type(() => Date)
    @CreateDateColumn()
    @ApiProperty({ type: Date, format: 'date-time' })
    declare readonly createdAt?: Date;

    @IsDateString(undefined, {
        groups: [GROUP.READ_MANY, GROUP.READ_ONE],
    })
    @IsOptional({ groups: [GROUP.READ_MANY, GROUP.READ_ONE] })
    @UpdateDateColumn()
    @Type(() => Date)
    @ApiProperty({ type: Date, format: 'date-time' })
    declare readonly updatedAt?: Date;

    @Exclude()
    @IsOptional({ always: true })
    @DeleteDateColumn({
        type: 'timestamp',
        nullable: true,
    })
    @IsDateString(undefined, {
        groups: [GROUP.READ_MANY, GROUP.READ_ONE, GROUP.SEARCH, GROUP.PARAMS],
    })
    @Type(() => Date)
    @ApiHideProperty()
    declare readonly deletedAt?: Date;
}
`
yes, it is defined, also, i dont know if this was the behavior before, but update seems to be inserting a new document

**Comment by jiho-kr** [comment]
`WHERE "ClientEntity"."deleted_at" IS NULL` is generated by typeorm (withDeleted option).

Would you like to check it without CRUD library?
```
this.repository.findOne({ withDeleted: true });
```

**Comment by jiho-kr** [comment]
https://github.com/woowabros/nestjs-library-crud/releases/tag/0.11.1

**Comment by chihabhajji** [comment]
🙌🙌
